lib/metasploit/framework/login_scanner/teamcity.rb

```diff
@@ -209,18 +226,21 @@ def try_login(username, password, public_key, retry_counter = 0)
           # Currently, those building blocks are not available, so this is the approach I have implemented.
           timeout = res.body.match(/login only in (?<timeout>\d+)s/)&.named_captures&.dig('timeout')&.to_i
           if timeout
-            framework_module.print_status "User '#{username}' locked out for #{timeout} seconds. Sleeping, and retrying..."
-            sleep(timeout + 1) # + 1 as TeamCity is off-by-one when reporting the lockout timer.
-            result = try_login(username, password, public_key, retry_counter + 1)
-            return result
+            logger.print_status "#{@host}:#{@port} - User '#{username}:#{password}' locked out for #{timeout} seconds. Sleeping, and retrying..." if logger
```

Comment:
```suggestion
            framework_module.print_status "#{@host}:#{@port} - User '#{username}:#{password}' locked out for #{timeout} seconds. Sleeping, and retrying..." if framework_module
```

I think we should only need the `framework_module` here 🤔